src-cpp/ProducerImpl.cpp

```diff
@@ -143,14 +143,19 @@ RdKafka::ProducerImpl::produce (RdKafka::Topic *topic,
 
 }
 
-
 RdKafka::ErrorCode
 RdKafka::ProducerImpl::produce (const std::string topic_name,
                                 int32_t partition, int msgflags,
                                 void *payload, size_t len,
                                 const void *key, size_t key_len,
-                                int64_t timestamp,
-                                void *msg_opaque) {
+                                int64_t timestamp, void *msg_opaque,
+                                RdKafka::Headers *headers) {
+  rd_kafka_headers_t *hdrs;
+  if (headers) {
+    hdrs = headers->c_headers();
+  } else {
+    hdrs = 0;
```

Comment:
use NULL for pointers, not 0. And in this case you can initialize `hdrs` to NULL directly.